plugin/semisync/semisync_slave_plugin.cc

```diff
@@ -129,7 +130,23 @@ static int repl_semi_slave_read_event(Binlog_relay_IO_param *,
 
 static int repl_semi_slave_queue_event(Binlog_relay_IO_param *param,
                                        const char *, unsigned long, uint32) {
+  DBUG_EXECUTE_IF("before_semi_sync_event", {
```

Comment:
Can we use `before_semi_sync_reply` below instead of creating a new sync point?

Reply:
`before_semi_sync_event` is the same as your `wait_before_semi_sync_reply`.

We can replace `before_semi_sync_event` with `before_semi_sync_reply` and MTR test should work, but the behavior is different:
a) `before_semi_sync_event` - An unacked trx was not yet commited on a semi-sync slave
b) `before_semi_sync_reply` - An unacked trx was already commited on a semi-sync slave